synapse/third_party_rules/access_rules.py

```diff
@@ -542,13 +542,14 @@ def _get_join_rule_from_state(state_events):
         return join_rule_event.content.get("join_rule")
 
     @staticmethod
-    def _get_members_and_tokens_from_state(state_events):
+    def _get_members_and_tokens_from_state(state_events, event):
         """Retrieves from a list of state events the list of users that have a
         m.room.member event in the room, and the tokens of 3PID invites in the room.
 
         Args:
             state_events (dict[tuple[event type, state key], EventBase]): The set of state
                 events.
+            event (EventBase): The event being checked.
```

Comment:
the docstring on the function seems to describe an operation that happens on the room, rather than a particular event, so it's hard to understand what `event` means here.

TBH I think that's an indication that you're adding functionality that doesn't belong in this function. I mean, you could add a parameter `exclude_events_for_user` which might fit better, but I'm not clear why that's the right thing to do.